mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -508,12 +508,10 @@ class LogisticRegressionModel private[spark] (
    * thrown if `trainingSummary == None`.
    */
   @Since("1.5.0")
-  def summary: LogisticRegressionTrainingSummary = trainingSummary match {
-    case Some(summ) => summ
-    case None =>
-      throw new SparkException(
-        "No training summary available for this LogisticRegressionModel",
-        new NullPointerException())
+  def summary: LogisticRegressionTrainingSummary = trainingSummary.getOrElse {
+    throw new SparkException(
+      "No training summary available for this LogisticRegressionModel",
+      new RuntimeException())
```

Comment:
Generally you don't use `RuntimeException` directly... why is the exception chained anyway and what was wrong with NPE? just kinda inaccurate? `NoSuchElementException` is what `None.get` throws.